model-optimizer/extensions/front/tf/if_ext.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+
+from extensions.front.tf.while_ext import update_body_graph
+from extensions.ops.If import If
+from extensions.ops.parameter import Parameter
+from mo.front.common.register_custom_ops import check_for_duplicates
+from mo.front.extractor import FrontExtractorOp, extract_node_attrs
+from mo.front.tf.extractor import tf_op_extractor, tf_op_extractors
+from mo.front.tf.extractors.utils import tf_dtype_extractor
+from mo.graph.graph import Node, Graph
+from mo.ops.op import PermuteAttrs
+
+
+def extract_method(cls, if_node: Node):
+    If.update_node_stat(if_node, {})
+
+    if_name = if_node.soft_get('name', if_node.id)
+
+    # check that required body and condition functions exist in the graph library
+    main_graph = if_node.graph
+    then_graph_name = if_node.pb.attr['then_branch'].func.name
+    else_graph_name = if_node.pb.attr['else_branch'].func.name
+    assert 'library' in main_graph.graph, 'The graph does not contain a library that is required ' \
+                                          'by node with name "{}".'.format(if_name)
+    library_graph = main_graph.graph['library']
+
+    assert then_graph_name in library_graph, 'The library does not contain a function with name "{}" ' \
+                                             'that is required by node ' \
+                                             'with name "{}".'.format(then_graph_name, if_name)
+    then_graph_proto = library_graph[then_graph_name]
+
+    assert else_graph_name in library_graph, 'The library does not contain a function with name "{}" ' \
+                                             'that is required by node ' \
+                                             'with name "{}".'.format(else_graph_name, if_name)
+    else_graph_proto = library_graph[else_graph_name]
```

Comment:
I propose to think of avoiding this code duplicates  and create util function for this. It is a copy-paste from While_ext

Reply:
done